primaryHandler.go

```diff
@@ -51,99 +57,211 @@ const (
 )
 
+// Configuration file dot-delimited paths for convenience and protection against typos
+const (
+	//JWTValidatorConfigKey is the path to the JWT
+	//validator config for device registration endpoints
+	JWTValidatorConfigKey = "jwtValidator"
+
+	//DeviceAccessCheckConfigKey is the path to the validator config for
+	//restricting API access to devices based on known device metadata and credentials
+	//presented by API consumers
+	DeviceAccessCheckConfigKey = "deviceAccessCheck"
+
+	//ServiceBasicAuthConfigKey is the path to the list of accepted basic auth keys
+	//for the API endpoints (note: does not include device registration)
+	ServiceBasicAuthConfigKey = "inbound.authKey"
+
+	//InboundTimeoutConfigKey is the path to the request timeout duration for
+	//requests inbound to devices connected to talaria
+	InboundTimeoutConfigKey = "inbound.timeout"
+)
+
+//NoOpConstructor provides a transparent way for constructors that make up
+//our middleware chains to work out of the box even without configuration
+//such as authentication layers
+var NoOpConstructor = func(h http.Handler) http.Handler { return h }
+
+//JWTValidator provides a convenient way to define jwt validator through config files
 type JWTValidator struct {
 	// JWTKeys is used to create the key.Resolver for JWT verification keys
 	Keys key.ResolverFactory `json:"keys"`
 
-	// Custom is an optional configuration section that defines
-	// custom rules for validation over and above the standard RFC rules.
-	Custom secure.JWTValidatorFactory `json:"custom"`
+	// Leeway is used to set the amount of time buffer should be given to JWT
+	// time values, such as nbf
+	Leeway bascule.Leeway
 }
 
 func getInboundTimeout(v *viper.Viper) time.Duration {
-	if t, err := time.ParseDuration(v.GetString("inbound.timeout")); err == nil {
+	if t, err := time.ParseDuration(v.GetString(InboundTimeoutConfigKey)); err == nil {
 		return t
 	}
 
 	return DefaultInboundTimeout
 }
 
-func NewPrimaryHandler(logger log.Logger, manager device.Manager, v *viper.Viper, a service.Accessor, e service.Environment, controlConstructor func(http.Handler) http.Handler) (http.Handler, error) {
+func SetLogger(logger log.Logger) func(delegate http.Handler) http.Handler {
+	return func(delegate http.Handler) http.Handler {
+		return http.HandlerFunc(
+			func(w http.ResponseWriter, r *http.Request) {
+				r = r.WithContext(logging.WithLogger(r.Context(),
+					log.With(logger, "requestHeaders", r.Header, "requestURL", r.URL.EscapedPath(), "method", r.Method)))
+				delegate.ServeHTTP(w, r)
+			})
+	}
+}
+
+func GetLogger(ctx context.Context) bascule.Logger {
+	logger := log.With(logging.GetLogger(ctx), "ts", log.DefaultTimestampUTC, "caller", log.DefaultCaller)
+	return logger
+}
+
+//buildUserPassMap decodes base64-encoded strings of the form user:pass and write them to a map from user -> pass
+func buildUserPassMap(logger log.Logger, encodedBasicAuthKeys []string) (userPass map[string]string) {
```

Comment:
I think we should put this in the `basculehttp` package.

Reply:
Probably without the logger as a parameter...

Reply:
Yeah, that sounds like a good idea. I added this issue: https://github.com/xmidt-org/bascule/issues/57